File: replication_files/urban_accounting_welfare_replication/ReplicationFiles/MatlabPro/NsysEFOPA.m
function D = NsysEFOPA(X)

global i shocks kappa psi ubarEFO theta inter eps RTFP AA ksi

  N = X;  
    
  c1 = ubarEFO + (1 + psi)*log((1 + psi))-psi*log(psi)-AA*(N^ksi);
  c2 = (1-theta)*(((RTFP*(N^eps))^(1/(1-theta)))/((theta/inter)^(theta/(1-theta))));
  
  D(i) = (pi/(kappa^2))*((log(c2)-c1)/(((1+psi)/c2)+(2/3)*exp(shocks(i,4))))^2-N;

   
D = [D(i)];
        

File: src/NsysEFOPA.jl
module NsysEFOPA_

  export NsysEFOPA

  function NsysEFOPA(X) 

    h1 = ubarEFO + (1 + psi)*log((1 + psi))-psi*log(psi)-AA*(X^ksi)
    h2 = (1-theta)*(((RTFP*(X^eps))^(1/(1-theta)))/((theta/inter)^(theta/(1-theta))))
    
    f = (pi/(kappa^2))*((log(h2)-h1)/(((1+psi)/h2)+(2/3)*exp(shocks[i,4])))^2-X
    

    return f

  end             
    
  
end #End NsysEFOPA